project/src/components/offer-card/offer-card.tsx

```diff
@@ -1,29 +1,66 @@
 import { Link } from 'react-router-dom';
+import { Offer } from '../../types/offer';
+import { capitalize } from '../../utils/common';
+import { AppRoute } from '../../const';
+
+type OfferCardProps = {
+  offer: Offer;
+  extraClass: string;
+  onMouseEnter?: (id: Offer['id']) => void;
+  onMouseLeave?: () => void;
+  isSmall?: boolean;
+};
+
+function OfferCard({ offer, onMouseEnter, onMouseLeave, extraClass, isSmall }: OfferCardProps) {
+  const {
+    id,
+    isPremium,
+    isFavorite,
+    previewImage,
+    title,
+    price,
+    rating,
+    type,
+  } = offer;
+
+  const capitalizedType = capitalize(type);
+  const offerPageUrl = AppRoute.OfferBase + id;
```

Comment:
Мне кажется можно не выносить и записать прямо в линк так. Либо ты можешь собирать путь через функцию generatePath 
```suggestion
  ={`${AppRoute.OfferBase}/${id}`};
```